File: Front-End/src/app/DTO/card_product/CardProduct.dto.ts
import {SoftwareCardDTO} from "./SoftwareCard.dto";
import {AuthorizationPolicyDTO} from "./AuthorizationPolicyDTO";

export interface CardProductDTO {
    idCardProduct: number;
    name: string;
    cardType: string;
    contactlessUpperLimitPerTransaction: number;
    numberOfContactlessTransactionBeforeAskingPin: number;
    softwareCards: SoftwareCardDTO[];
    authorizationPolicies: AuthorizationPolicyDTO[];
}

File: Back-End/src/main/java/fr/ensicaen/pi/gpss/backend/database/dto/card_product/CardProductDTO.java
package fr.ensicaen.pi.gpss.backend.database.dto.card_product;

import com.fasterxml.jackson.annotation.JsonInclude;
import fr.ensicaen.pi.gpss.backend.database.dto.TemplateDTO;
import fr.ensicaen.pi.gpss.backend.database.enumerate.CardType;
import jakarta.validation.Valid;
import jakarta.validation.constraints.*;
import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Builder
@Getter
@JsonInclude(JsonInclude.Include.NON_NULL)
@Valid
public class CardProductDTO extends TemplateDTO {
    @PositiveOrZero
    private Integer idCardProduct;
    @NotBlank
    private String name;
    private CardType cardType;
    @Positive
    private Integer contactlessUpperLimitPerTransaction;
    @Positive
    private Integer numberOfContactlessTransactionBeforeAskingPin;
    @NotEmpty
    private List<SoftwareCardDTO> softwareCards;
    @NotEmpty
    private List<AuthorizationPolicyDTO> authorizationPolicies;
}
